src/rank_llm/rerank/reranker.py

```diff
@@ -1,187 +0,0 @@
-from datetime import datetime
-from pathlib import Path
-from typing import List
-
-from tqdm import tqdm
-
-from rank_llm.data import DataWriter, Request, Result
-from rank_llm.rerank.rankllm import RankLLM
-
-
-class Reranker:

```

Comment:
IIUC, you are removing the Reranker class and add rerank and rerank_batch to the RankLLM class. 
hm, I am not sure if we want this. I kind of like a Reranker class that has a ranking agent. For example, writing the reranking ressults can be a task of the reranker, but not the ranking agent.
WDYT? @ronakice

Reply:
We might also have Rerankers with other agents. For example, our Identity and Random rerankers do not inherit from rankllm but still need to implement the rerank and rerank_batch methods.